### PR TITLE
KafkaSource: Fixed wrong error logging during reconciliation

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

From https://github.com/knative/eventing-contrib/issues/1291#issuecomment-638221293

### CODE REVIEW

**Comment by slinkydeveloper** [comment]
Seems like it's a consistent failure on kafkasource tests aka i broke something :)

**Comment by aliok** [comment]
@slinkydeveloper Or the tests are problematic.

File: kafka/source/pkg/reconciler/source/kafkasource.go [hunk 1]
```diff
@@ -135,8 +135,14 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, src *v1alpha1.KafkaSourc
...
 	if err != nil {
-		logging.FromContext(ctx).Error("Unable to create the receive adapter", zap.Error(err))
-		return err
+		var event *pkgreconciler.ReconcilerEvent
+		isReconcilerEvent := pkgreconciler.EventAs(err, &event)
+		if isReconcilerEvent && event.EventType != corev1.EventTypeNormal {
+			logging.FromContext(ctx).Error("Unable to create the receive adapter. Reconciler error", zap.Error(err))
+			return err
+		} else if !isReconcilerEvent {
+			logging.FromContext(ctx).Error("Unable to create the receive adapter. Generic error", zap.Error(err))
```
Comment (aliok) [inline kafka/source/pkg/reconciler/source/kafkasource.go:144]: @slinkydeveloper shouldn't we return err here?

Comment (slinkydeveloper) [reply kafka/source/pkg/reconciler/source/kafkasource.go:144]: Uhm yes we should :smile: Mind doing a PR?

Comment (aliok) [reply kafka/source/pkg/reconciler/source/kafkasource.go:144]: https://github.com/knative/eventing-contrib/pull/1318

**Pull request merged**